### PR TITLE
feat: sort options on explore

### LINKED ISSUE #4825: `explore` sort options near tabs
Capture d’écran 2023-01-25 à 11 55 29 AM

In `ExploreTabsFilterSort` add sorting options near explore-tabs

### spec
use/edit `NeoSelect` component

### note
There will be different sorting options between `COLLECTION` & `ITEMS`

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #4825

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

Screenshot 2023-01-27 at 20 04 24

### CODE REVIEW

File: components/explore/ExploreSort.vue [hunk 1]
```diff
@@ -0,0 +1,125 @@
+<template>
+  <div>
+    <o-dropdown
```
Comment (roiLeo) [inline components/explore/ExploreSort.vue:3]: `<NeoDropdown>`?

File: components/explore/ExploreSort.vue [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+const sortOptions = ref<string[]>([])
+const selectedSort = computed({
+  get() {
+    return sortOptions.value
+  },
+  set(value) {
+    sortOptions.value = selectiveSort(value)
+  },
+})
```
Comment (roiLeo) [inline components/explore/ExploreSort.vue:69]:
```suggestion
const selectedSort = computed({
  get: () => sortOptions.value,
  set: (value) => sortOptions.value = selectiveSort(value),
})
```

File: components/explore/ExploreSort.vue [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+import { NFT_SQUID_SORT_CONDITION_LIST } from '@/utils/constants'
+
+const route = useRoute()
+const router = useRouter()
+const { $i18n } = useNuxtApp()
+
+const options = NFT_SQUID_SORT_CONDITION_LIST
+
+function selectiveSort(options) {
```
Comment (roiLeo) [inline components/explore/ExploreSort.vue:41]: `options` type

File: components/explore/ExploreSort.vue (hunk 1 above)
Comment (vikiival) [reply components/explore/ExploreSort.vue:3]: + 1 for neodropdown

File: styles/abstracts/_theme.scss [hunk 4]
```diff
@@ -29,7 +29,7 @@ $themes: (
     'k-hovergrey': #6b6b6b,
     'k-blue': #6188e7,
     'k-shade': #cccccc,
-    'primary-shadow': 4px 4px $white,
+    'primary-shadow': 4px 4px white,
```
Comment (vikiival) [inline styles/abstracts/_theme.scss:32]: Why?

File: components/explore/ExploreTabs.vue [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        class="explore-tabs-button"
+        tag="nuxt-link"
+        :active="selectedTab === TabType.COLLECTION"
+        to="collectibles">
+        <span> {{ $t('collections') }}</span>
+        <img
+          v-if="selectedTab === TabType.COLLECTION"
+          src="/checkmark.svg"
+          class="check-icon" />
+      </NeoButton>
```
Comment (vikiival) [inline components/explore/ExploreTabs.vue:27]: `<ExploreTabItem />`

File: components/shared/gallery/Gallery.vue [hunk 6]
```diff
@@ -98,9 +99,10 @@ export default class Gallery extends mixins(
   private searchQuery: SearchQuery = {
     search: this.$route.query?.search?.toString() ?? '',
     type: this.$route.query?.type?.toString() ?? '',
-    sortByMultiple: this.$route.query?.sort?.toString()
-      ? [this.$route.query?.sort?.toString()]
-      : undefined,
+    sortByMultiple:
+      typeof this.$route.query?.sort === 'string'
+        ? [this.$route.query?.sort]
+        : this.$route.query?.sort,
```
Comment (vikiival) [inline components/shared/gallery/Gallery.vue:105]: isnt it always string?

**Review by vikiival — CHANGES_REQUESTED** [review]
Smol

**Comment by Jarsen136** [comment]
WFM and I would like to make the option to stay on one line

image

**Comment by yangwao** [comment]
Probably we can remove duplicate sort in menu below?
image

File: components/explore/ExploreSort.vue (hunk 1 above)
Comment (roiLeo) [reply components/explore/ExploreSort.vue:3]: `@click.native="onChange"` on NeoDropdownItem maybe? but that doesn't look right multiple selection won't work
Or maybe use `NeoSelect` multiple instead

**Comment by roiLeo** [comment]
By the way could you change back the tabs order "Collections > Items" we have already mentioned this in a discussion and it hasn't been updated on figma yet

**Comment by exezbcz** [comment]
good! 
image
- please add small check to selected options
- the number of filter - to pink color as in light mode. 
- align the "sort by" text to left
- when you click on sort by and the dropdown is displayed, change the button to active state. as on gallery item cards (chart)

image

File: components/explore/ExploreSort.vue (hunk 1 above)
Comment (roiLeo) [reply components/explore/ExploreSort.vue:3]: I'm fine with it, I don't really see the point of multiple sorting options

**Review by roiLeo — COMMENTED** [review]
Issue on responsive
Screenshot 2023-01-27 at 16-42-04 Low minting fees and carbonless NFTs KodaDot - NFT Market Explorer

Dropdown caret should change on open?

Screenshot 2023-01-27 at 16-46-39 Low minting fees and carbonless NFTs KodaDot - NFT Market Explorer

Do we really need to set a fixed width on "Sort by" button?

Latter (new issue) I would add Sort button on Collections too (New first, Old first)

**Comment by exezbcz** [comment]
> > Do we really need to set a fixed width on "Sort by" button?
> 
> cc @exezbcz on figma **162px** currently, 10rem = **160px** if not set the fixed width, **108px**
> 
> > I would add Sort button on Collections too (New first, Old first)
> 
> I think this is still feasible, let me add in this PR

it does not have to fixed. 
image
- its bit larger because I wanted to stretch it to the whole page width - thus making it bigger and easier to click. 
- important is that it has 16px 8px padding.

**Review by Jarsen136 — APPROVED** [review]
code lgtm

**Comment by exezbcz** [comment]
the notif count to k-accent in dark mode
image
- also it does not look like its center aligned 

other than that it seems good to me

**Comment by exezbcz** [comment]
> > the notif count to k-accent in dark mode image
> > 
> > * also it does not look like its center aligned
> > 
> > other than that it seems good to me
> 
> updated, accent color and centering

image
- I meant the bg color, so its the same as in light mode :DDD

**Review by roiLeo — APPROVED** [review]
✅ nice! wfm

**Pull request merged**